Use message type instead of prototype

0x26res opened the pull request:
I've fixed the issue and added some tests. 

But I wasn't reproduce the issue with this repo set up. The issue stems from the fact that in some environment:
```
MessageFactory().GetPrototype(message_type.DESCRIPTOR) is not messsage_type
```

I think it must have something to do with a different combination of `protoc` and `protobuf` python library.

Does anyone happen to know what version of `protoc` has been used to generate the pb2.py files?

Thanks.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/confluentinc/confluent-kafka-python?pullRequest=1407) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/confluentinc/confluent-kafka-python?pullRequest=1407) before we can accept your contribution.<br/><hr/>**0x26res** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/confluentinc/confluent-kafka-python?pullRequest=1407) it.</sub>

pranavrth commented on tests/integration/schema_registry/test_proto_serializers.py:
```diff
@@ -23,30 +23,10 @@
     PublicTestProto_pb2
 from tests.integration.schema_registry.data.proto.DependencyTestProto_pb2 import DependencyMessage
 from tests.integration.schema_registry.data.proto.exampleProtoCriteo_pb2 import ClickCas
+from tests.schema_registry.test_proto import TEST_DATA
```
It's better to keep data in their respective test files. Its fine here to replicate data in each file if we require that.

pranavrth commented on tests/schema_registry/test_proto.py:
```diff
@@ -19,13 +19,41 @@
+                                                          NestedTestProto_pb2)
+from tests.schema_registry.test_api_client import TEST_URL
 
+TEST_DATA = [
```
No need for such complex data while testing our usecase. Keep only what is really required.

Move data near the testcase.

pranavrth commented on tests/integration/schema_registry/test_proto_serializers.py:
```diff
@@ -117,7 +97,6 @@ def test_protobuf_serializer_type_mismatch(kafka_cluster):
                              r" 'tests.integration.schema_registry.data.proto.NestedTestProto_pb2.NestedMessage'\>"):
         producer.produce(topic, key=pb2_2())
 
-
```
Keep this extra line. Each test is separated by 2 lines.

pranavrth commented on src/confluent_kafka/schema_registry/protobuf.py:
```diff
@@ -510,7 +510,8 @@ def __init__(self, message_type, conf=None):
         self._index_array = _create_index_array(descriptor)
-        self._msg_class = MessageFactory().GetPrototype(descriptor)
+        self._msg_class = message_type
+
```
Remove this extra line.

pranavrth commented on tests/schema_registry/test_proto.py:
```diff
@@ -83,3 +111,27 @@ def test_index_encoder(msg_idx, zigzag, expected_hex):
+
+
+@pytest.mark.parametrize("pb2, data", TEST_DATA)
+def test_round_trip(mock_schema_registry, pb2, data):
```
Please write a better name for the test which suggests what we are trying to test. Something in the line of serializing, deserializing and then again serializing. Better to have verbose test name.

pranavrth commented on tests/schema_registry/test_proto.py:
```diff
@@ -83,3 +111,27 @@ def test_index_encoder(msg_idx, zigzag, expected_hex):
+    bytes_before = serializer(message_before, context)
+    message_after = deserializer(bytes_before, context)
+    bytes_after = serializer(message_after, context)
+    assert isinstance(message_after, pb2)
```
Keep all asserts continuously.

Add a line break before assert.

pranavrth commented on tests/schema_registry/test_proto.py:
```diff
@@ -83,3 +111,27 @@ def test_index_encoder(msg_idx, zigzag, expected_hex):
+    assert bytes_after == bytes_before
+
+
+def test_foo_bar():
```
We are removing this code altogether, so I think we don't need to test this. Please remove this function.

pranavrth reviewed (changes requested):
Please fix the build issues. Mostly related to the styling. 

https://app.travis-ci.com/github/confluentinc/confluent-kafka-python/builds/254834885#L827

0x26res commented:
@pranavrth thanks for the feed back. I've addressed the comments.

The CI is failing, but (unless I'm mistaken) I think the serdes branch has been left in a bad state, some tests need fixing.

pranavrth commented:
Hello @0x26res,

The testcases related to protobuf were failing in **serdes** branch. I have fixed and merged the test cases. Please resolve the conflict and it should work fine.

0x26res commented:
@pranavrth thanks for fixing, it looks good now.

pranavrth commented:
LGTM

0x26res commented:
@pranavrth I don't have write access so I won't be able to merge. Can I leave this to you? Thanks

0x26res commented:
@mhowlett is that not going through? Should I send another MR?
